app/src/lib/stores/pull-request-coordinator.ts

```diff
@@ -156,6 +156,33 @@ export class PullRequestCoordinator {
     }
   }
 
+  /**
+   * Get the last time a repository's pull requests were fetched
+   * from the GitHub API
+   *
+   * Since `PullRequestStore` stores these timestamps by
+   * GitHubRepository, we gather the timestamps for all
+   * **related repos** and return _the least recent one._
+   *
+   * If we can't find any timestamps stored, returns `undefined`
+   */
+  public getLastRefreshed(
+    repository: RepositoryWithGitHubRepository
+  ): number | undefined {
+    const ghRepos = [repository.gitHubRepository]
+    if (repository.gitHubRepository.parent !== null) {
+      ghRepos.push(repository.gitHubRepository.parent)
+    }
+    const times = new Array<number>()
+    for (const ghr of ghRepos) {
+      const lastRefreshed = this.pullRequestStore.getLastRefreshed(ghr)
+      if (lastRefreshed !== undefined) {
+        times.push(lastRefreshed)
+      }
+    }
+    return times.length > 0 ? Math.min(...times) : undefined
```

Comment:
This method took me a pass or two to grok. I wonder if we could make it easier to follow if we laid it out sequentially and got rid of the intermediate arrays to emphasize that there's max two repositories being checked. Something like

```ts
const ghr = repository.gitHubRepository
const lastRefresh = this.pullRequestStore.getLastRefreshed(ghr)

const parentLastRefresh = ghr.parent
  ? this.pullRequestStore.getLastRefreshed(ghr.parent)
  : undefined

return !lastRefresh || !parentLastRefresh
  ? lastRefresh || parentLastRefresh
  : Math.min(lastRefresh, parentLastRefresh)
```

Reply:
i like this! thank you! da7657a2a6cb88ea4ede75e2e75c2f503f8501b5